tr2: add support for The Golden Mask

lahm86 opened the pull request:
Resolves #1554.
Resolves #1604.
Resolves #1621.
Resolves #2667.
Resolves #2677.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

The Golden Mask can now be launched by using `-gold` on the command line, as well as from within the config tool. The data files are expected to be named appropriately per the game flow. I will share the zip for initial testing before we make it available through the installer. The installer will follow in its own issue/PR as it is more involved; I will also follow up separately with the remaining data issues, such as 3D pickup scaling.

To use legacy GM saves, they must be named with the format `savegame_gm.%d` as this is what the game flow is configured to look for, and is required because there is otherwise no difference between these and vanilla saves. The installer can perhaps deal with this and provide an option to import and rename legacy GM saves.

Testing
- The game should flow as expected, so collect all secrets and you get the credits and then Vegas, otherwise credits and exit to title. Note this differs from OG, as it would set a flag to make level select available and you would have to manually select Vegas, but for TRX we want to slot it into the existing bonus level support.
- Strings should be accurate (included the change in Ice Palace to White Tiger per #2667).
- Bears and wolves should behave as expected; monk type 1 should have no shadow
- In regular TR2, spiders, big spiders and monk type 1 should behave/appear normally
- Note that the `fix_bear_ai` config option is now available in TR2 as well, so getting the bear to pat attack Lara should work

Thanks @aredfan for getting so much of the specifics gathered prior to this. And thanks again @rr- for the object alias code.

lahm86 committed:
tr2: add support for The Golden Mask

This implements support for the Golden Mask by providing relevant game
flow files, command line handling, and object alias setup on launch.

Resolves #1554.
Resolves #1604.
Resolves #1621.
Resolves #2667.

lahm86 committed:
tr2/objects/monk: create Golden Mask monk alias

This allows the Golden Mask mod to refer to an alias of Monk1 as Monk3,
which in turn allows for the object's shadow to be removed.

lahm86 committed:
objects/bear: create Golden Mask bear alias

This allows for the big spider in the Golden Mask to be used as an
alias for the bear.

lahm86 committed:
objects/wolf: create Golden Mask wolf alias

This allows for the small spider in the Golden Mask to be used as an
alias for the wolf.

rr- commented on data/tr2/ship/cfg/TR2X_strings.json5:
```diff
@@ -141,7 +141,7 @@
             "title": "Ice Palace",
             "objects": {
-                "tiger": {"name": "Snow Leopard"},
+                "tiger": {"name": "White Tiger"},
```
White Tiger? :D

lahm86 replied:
aredfan shrewdly noticed this, so Ice Palace uses different textures compared to the Bengal tiger and snow leopard elsewhere 
![image](https://github.com/user-attachments/assets/7a954c85-91cf-48f4-bc78-6c292eb11100)

rr- commented on data/tr2/ship/cfg/TR2X_strings_gm.json5:
```diff
@@ -0,0 +1,474 @@
+    ],
+
+    "objects": {
+        "small_medipack":        {"name": "Small Medipack"},
```
No need to provide all the defaults- they're out there in the exe and in the main GF file as means of showing what can be changed.

rr- commented on src/tr2/game/objects/creatures/monk.c:
```diff
@@ -259,5 +249,24 @@ static void M_Control(const int16_t item_num)
+    if (!obj->loaded) {
+        return;
+    }
+    Monk1_Setup(obj);
```
Let's call here the base function and duplicate the pivot length assignment for extra visibility. I really want us to avoid inter-object setup calls as much as possible.

aredfan commented on data/tr2/ship/cfg/TR2X_strings_gm.json5:
```diff
@@ -0,0 +1,61 @@
+            "title": "Furnace of the Gods",
+            "objects": {
+                "big_spider": {"name": "Polar Bear"},
+                "spider": {"name": "Wolf"},
+                "puzzle_1": {"name": "Mask Of Tornarsuk"},
+                "puzzle_2": {"name": "Gold Nugget"},
```
Lower case for `of` in `Mask Of Tornarsuk`

aredfan commented on data/tr2/ship/cfg/TR2X_strings_gm.json5:
```diff
@@ -0,0 +1,61 @@
+        {
+            "title": "Kingdom",
+            "objects": {
+                "tiger": {"name": "Snow Leopard"},
+                "puzzle_1": {"name": "Mask Of Tornarsuk"},
+            },
```
Lower case for `of` in `Mask Of Tornarsuk`

aredfan commented on data/tr2/ship/cfg/TR2X_strings_gm.json5:
```diff
@@ -0,0 +1,61 @@
+        },
+
+        // 5. Nightmare in Vegas
+        {
+            "title": "Nightmare In Vegas",
+            "objects": {
```
Lower case for `in` in `Nightmare In Vegas`

aredfan reviewed (commented):
Just a quick bit of proof-reading before I get started with the testing.

aredfan reviewed (changes requested):
I found just one issue during testing - the `Mask of Tornarsuk` will need to be automatically added to Lara's inventory at the start of Kingdom.

**Edit:** This might be another issue, when you start Nightmare in Vegas, there's no pistols visible in Lara's holsters. This issue is probably already covered in #1588.

lahm86 committed:
tr2/inventory: handle GF weapon removal and addition

This responds to scenarios when the game flow removes all weapons but
also re-adds pistols, such that Lara's meshes can be drawn correctly.

Resolves #2677.

lahm86 commented:
Thanks @aredfan, everything should now be addressed. Re the pistols, this is a specific GF bug, so fixed here under its own issue.

aredfan reviewed (approved):
@lahm86 thank you for all your hard work in making the GM port a reality.

LGTM. 👍

walkawayy commented on data/tr2/ship/cfg/TR2X_gameflow_gm.json5:
```diff
@@ -0,0 +1,137 @@
+                {"type": "display_picture", "path": "data/credit07_gm.pcx", "display_time": 15, "fade_in_time": 0.5, "fade_out_time": 0.5},
+                {"type": "display_picture", "path": "data/credit08.pcx", "display_time": 15, "fade_in_time": 0.5, "fade_out_time": 0.5},
+                {"type": "total_stats", "background_path": "data/end.pcx"},
+                {"type": "level_complete"},
```
I think we forgot a
`{"type": "level_stats"},`
here? Upon Kingdom's completion, no level stats are shown before the credits.

lahm86 replied:
Thanks, this is how it's configured in the original though. HSH is similar, so it goes straight to credits.

walkawayy replied:
Do you think we should change it? It's kinda weird not to have stats.

Also semi related but a different issue idea, it would be nice to give some kind of visual indicator in the final stats about missing secrets causing you to miss out on the bonus level. The only way you know it exists is from Googling or other fans.

lahm86 replied:
I feel it's fine without, people can easily add it if they want. 

Having an indicator sounds good. Custom levels could benefit too.

rr- replied:
I'd opt for adding it but in a way that plays the skidoo theme already. A separate effort by all means.

lahm86 replied:
@walkawayy would you mind opening the separate tickets if you get a chance? Hopefully then we can draw a line under this PR.

walkawayy reviewed (changes requested):
LGTM except one gameflow issue.
